packages/skill-api/src/core/services/process-stripe-webhook.ts

```diff
@@ -285,25 +285,40 @@ export const processStripeWebhook = async (
       },
     })
 
-    const user = merchantCustomer?.user
+    const currentUser = merchantCustomer?.user
 
-    if (user) {
-      const currentEmail = user.email
-      const {email, name} = eventObject
+    if (currentUser) {
+      const {email: targetEmail, name} = z
+        .object({
+          email: z.string(),
+          name: z.string().optional(),
+        })
+        .parse(eventObject)
+
+      const {previous_attributes} = z
+        .object({
+          previous_attributes: z
+            .object({email: z.string().optional()})
+            .optional(),
+        })
+        .parse(event.data)
```

Comment:
We can pull the previous email out of `previous_attributes` that stripe includes in its webhook payload.